### PR TITLE
added auth implementation

### PR DESCRIPTION
**Pull request by szaszgyozo**

### CODE REVIEW

File: lib/router.dart [hunk 1]
```diff
@@ -1,75 +1,123 @@
...
+class AppRouter {
+  final _rootNavigatorKey = GlobalKey<NavigatorState>();
+  final _shellNavigatorKey = GlobalKey<NavigatorState>();
+
+  final LoginCubit loginCubit;
+  AppRouter(this.loginCubit);
+
+  late final router = GoRouter(
+      navigatorKey: _rootNavigatorKey,
+      initialLocation: '/home',
```
Comment (danutDiac) [inline lib/router.dart:23]: I propose to replace these [magic strings](https://deviq.com/antipatterns/magic-strings) with constants.

File: lib/main.dart [hunk 2]
```diff
@@ -30,11 +38,21 @@ class _AppScaffoldState extends State<AppScaffold> {
...
+            },
+          )
+        ],
+      ),
       body: widget.child,
       bottomNavigationBar: BottomNavigationBar(
-        currentIndex: calculateSelectedIndex(context),
-        onTap: (value) => onTap(value, context),
+        currentIndex: AppRouter.calculateSelectedIndex(context),
+        onTap: (value) => AppRouter.onTap(value, context),
```
Comment (danutDiac) [inline lib/main.dart:55]: `AppRouter.onTap` is generic (is not easy to figure out what it does unless you check the implementation). Can we chose a more subjective / intuitive name?

The final purpose would be to easily figure out what would happen on `onTap` function from the `BottomNavigationBar` without checking all the code.

Here's an example:
```
      bottomNavigationBar: BottomNavigationBar(
        ....
        onTap: (pageIndex) => navigateToPage(pageIndex, context),
```
These is just an example. You can pick other names that seem suggestive to you.

Comment (danutDiac) [reply lib/main.dart:55]: @ilie-lupu-ts @szaszgyozo

File: lib/router.dart (hunk 1 above)
Comment (danutDiac) [reply lib/router.dart:23]: @ilie-lupu-ts @szaszgyozo

Comment (szaszgyozo) [reply lib/router.dart:23]: Fixed in [this](https://github.com/ivoryio/SolarisDemoApp/pull/3) PR.

File: lib/main.dart (hunk 2 above)
Comment (szaszgyozo) [reply lib/main.dart:55]: Fixed in [this](https://github.com/ivoryio/SolarisDemoApp/pull/3) PR.

**Pull request merged**